### PR TITLE
Add PR template

### PR DESCRIPTION
**Pull request by nosheen-adil**
## Description 
Add a PR template to improve the review process for open source contributions.

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+- [ ] Refactor
+- [ ] Feature
+- [ ] Bug Fix
+- [ ] Optimization
+- [ ] Documentation Update
+
+## Description
+
+## Related Tickets & Documents
```
Comment (JohannCooper) [inline .github/pull_request_template.md:12]:
```suggestion
## Related Issues & Documents
```
To match the GitHub vocabulary and wording used below.

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
+
+## What type of PR is this? (check all applicable)
+
+- [ ] Refactor
+- [ ] Feature
+- [ ] Bug Fix
+- [ ] Optimization
+- [ ] Documentation Update
+
+## Description
```
Comment (JohannCooper) [inline .github/pull_request_template.md:10]:
```suggestion
## Description

_Please provide a brief description of the changes introduced by this pull
request._
```

Might be good to have a prompt here as well.

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+- Closes #
+
+## Testing Instructions
+
+_Please replace this line with instructions on how to test your changes, a note
+on the OS and python versions this has been tested on, as well as any relevant
+images/screenshots._
+
+## Added/updated tests?
+_We encourage you to write unit tests when possible._
```
Comment (JohannCooper) [inline .github/pull_request_template.md:24]:
```suggestion
<!-- We encourage you to write unit tests when possible. -->
```
Maybe we change this to a comment so contributors still see it when filling out the template but it doesn't appear when published. It also seems like the convention established here is that italic text should be replaced. This line isn't necessarily something to be replaced.

**Review by JohannCooper — CHANGES_REQUESTED** [review]
Thanks for making this! 

Just a few suggestions, open to other opinions.

**Review by Zachary-Ip — APPROVED** [review]
Thanks for adding this!

**Review by JohannCooper — APPROVED** [review]
Looks great!

**Pull request merged**